mlir/include/mlir/Dialect/MQTOpt/Transforms/Transpilation/Layout.h

```diff
@@ -10,135 +10,254 @@
+    return qubits_[this->programToHardware_[prog]];
+  }
+
+  /**
+   * @brief Check whether the layout contains a qubit.
+   * @param q The SSA Value representing the qubit.
+   * @return True if the layout contains the qubit, false otherwise.
+   */
+  [[nodiscard]] bool contains(const mlir::Value q) const {
+    return valueToMapping_.contains(q);
   }
 
   /**
    * @brief Replace an old SSA value with a new one.
    */
-  void remapQubitValue(const Value in, const Value out) {
+  void remapQubitValue(const mlir::Value in, const mlir::Value out) {
     const auto it = valueToMapping_.find(in);
-    assert(it != valueToMapping_.end() && "forward: unknown input value");
+    assert(it != valueToMapping_.end() &&
+           "remapQubitValue: unknown input value");
 
-    const QubitInfo map = it->second;
-    qubits_[map.hardwareIdx] = out;
+    const QubitInfo info = it->second;
+    qubits_[info.hw] = out;
 
     assert(!valueToMapping_.contains(out) &&
-           "forward: output value already mapped");
+           "remapQubitValue: output value already mapped");
 
-    valueToMapping_.try_emplace(out, map);
+    valueToMapping_.try_emplace(out, info);
     valueToMapping_.erase(in);
   }
 
   /**
    * @brief Swap the locations of two program qubits. This is the effect of a
    * SWAP gate.
    */
-  void swap(const Value q0, const Value q1) {
-    auto ita = valueToMapping_.find(q0);
-    auto itb = valueToMapping_.find(q1);
-    assert(ita != valueToMapping_.end() && itb != valueToMapping_.end() &&
+  void swap(const mlir::Value q0, const mlir::Value q1) {
+    auto it0 = valueToMapping_.find(q0);
+    auto it1 = valueToMapping_.find(q1);
+    assert(it0 != valueToMapping_.end() && it1 != valueToMapping_.end() &&
            "swap: unknown values");
-    std::swap(ita->second.programIdx, itb->second.programIdx);
-    std::swap(programToHardware_[ita->second.programIdx],
-              programToHardware_[itb->second.programIdx]);
+
+    const QubitIndex prog0 = it0->second.prog;
+    const QubitIndex prog1 = it1->second.prog;
+
+    std::swap(it0->second.prog, it1->second.prog);
+
+    ThinLayout::swap(prog0, prog1);
   }
 
   /**
    * @brief Return the current layout.
    */
-  ArrayRef<QubitIndex> getCurrentLayout() { return programToHardware_; }
+  mlir::ArrayRef<QubitIndex> getCurrentLayout() const {
+    return this->programToHardware_;
+  }
 
   /**
    * @brief Return the SSA values for hardware indices from 0...nqubits.
    */
-  [[nodiscard]] ArrayRef<Value> getHardwareQubits() const { return qubits_; }
+  [[nodiscard]] mlir::ArrayRef<mlir::Value> getHardwareQubits() const {
+    return qubits_;
+  }
 
 private:
   struct QubitInfo {
-    QubitIndex hardwareIdx;
-    QubitIndex programIdx;
+    QubitIndex prog;
+    QubitIndex hw;
   };
 
   /**
    * @brief Maps an SSA value to its `QubitInfo`.
    */
-  DenseMap<Value, QubitInfo> valueToMapping_;
+  mlir::DenseMap<mlir::Value, QubitInfo> valueToMapping_;
```

Comment:
I believe most, if not all, of the `mlir::` namespace specifiers could be safely dropped by adding back the `using namespace mlir`. Obviously just a fairly minor comment, but makes some of the lines shorter.